packages/react-components/src/hooks/useAlarms/hookHelpers/predicates.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  AlarmAssetModelRequest,
+  AlarmAssetRequest,
+  AlarmCompositeModelRequest,
+  AlarmInputPropertyRequest,
+  AlarmRequest,
+} from '../types';
+
+export const isAssetModelRequest = (
+  alarmRequest: AlarmRequest
+): alarmRequest is AlarmAssetModelRequest => Boolean(alarmRequest);
```

Comment:
Can we check the properties on the request that differentiate them? Like isAlarmAssetModelRequest is request.assetModelId != null && request.assetId == null or something similar.

Reply:
I should just explicitly check that `alarmRequest.assetId === undefined` for `isAssetModelRequest` and vice versa. We ensure that AlarmRequests can't get the fields mixed up between the assetId and assetModelId